.github/workflows/reusable-preflight.yml

```
@@ -98,34 +140,79 @@ jobs:
       # By default we use spot machines that can be terminated at any time.
       # Merge queues use persistent runners to avoid kicking off from queue when the runner is terminated.
       #
-      - id: set_runner
+      - name: Set runner
+        id: set_runner
         shell: bash
         run: |
+          echo "RUNNER_DEFAULT=parity-default" >> $GITHUB_OUTPUT
+          echo "RUNNER_WEIGHTS=parity-weights" >> $GITHUB_OUTPUT
+          echo "RUNNER_BENCHMARK=parity-benchmark" >> $GITHUB_OUTPUT
+          echo "RUNNER_MACOS=parity-macos" >> $GITHUB_OUTPUT
+          #
           # Run merge queues on persistent runners
           if [[ $GITHUB_REF_NAME == *"gh-readonly-queue"* ]]; then
-            echo "RUNNER=arc-runners-polkadot-sdk-beefy-persistent" >> $GITHUB_OUTPUT
-            echo "OLDLINUXRUNNER=oldlinux-persistent" >> $GITHUB_OUTPUT
+            echo "RUNNER=parity-large-persistent" >> $GITHUB_OUTPUT
+            echo "RUNNER_OLDLINUX=parity-oldlinux-persistent" >> $GITHUB_OUTPUT
           else
-            echo "RUNNER=arc-runners-polkadot-sdk-beefy" >> $GITHUB_OUTPUT
-            echo "OLDLINUXRUNNER=oldlinux" >> $GITHUB_OUTPUT
+            echo "RUNNER=parity-large" >> $GITHUB_OUTPUT
+            echo "RUNNER_OLDLINUX=parity-oldlinux" >> $GITHUB_OUTPUT
           fi
 
       #
       # Set vars
       #
-      - id: set_vars
+      - name: Set vars
+        id: set_vars
         shell: bash
         run: |
-          export BRANCH_NAME=${{ github.head_ref || github.ref_name }}
-          echo "SOURCE_REF_NAME=${BRANCH_NAME//\//-}" >> $GITHUB_OUTPUT
-          echo "COMMIT_SHA=${{ github.event.pull_request.head.sha || github.sha }}" >> $GITHUB_OUTPUT
+          export SOURCE_REF_NAME=${{ github.head_ref || github.ref_name }}
+          echo "SOURCE_REF_SLUG=${SOURCE_REF_NAME//\//_}" >> $GITHUB_OUTPUT
+          #
+          export COMMIT_SHA=${{ github.sha }}
+          echo "COMMIT_SHA=$COMMIT_SHA" >> $GITHUB_OUTPUT
+          echo "COMMIT_SHA_SHORT=${COMMIT_SHA:0:8}" >> $GITHUB_OUTPUT
+          #
+          export REF_NAME=${{ github.ref_name }}
+          echo "REF_SLUG=${REF_NAME//\//_}" >> $GITHUB_OUTPUT
 
 
       - name: log
         shell: bash
         run: |
           echo "workflow file: ${{ steps.current_file.outputs.currentWorkflowFile }}"
           echo "Modified: ${{ steps.set_changes.outputs.modified_keys }}"
+
+  #
+  #
+  #
+  ci-versions:
+    needs: [preflight]
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ needs.preflight.outputs.IMAGE }}
+    steps:
+
+      - name: Info rust
+        run: |
+          rustup show
+          cargo --version

```

```suggestion
          cargo +nightly --version
```
`rustup show` alone only shows what it would use by default, not what would be used in the current folder (since we plan on adding a rust-toolchain file).

---

added checkout step for `rustup show`